packages/ui/src/components/MaterialsCard/MaterialsCard.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { Box, BoxProps } from '@chakra-ui/react';
+
+import { Button } from '../Button';
+import { Typography } from '../Typography/Typography';
+
+type MaterialsCardStatus = 'idle' | 'loading' | 'generated';
+
+export interface MaterialsCardProps extends BoxProps {
+  status: MaterialsCardStatus;
+}
+
+type MaterialsCardOptions = { [status: string]: { statusText: string; buttonText: string } };
```

Comment:
💭 You can type it as `Record<MaterialsCardStatus, { statusText: string; buttonText: string }>`